akka-stream/src/main/scala/akka/stream/ActorMaterializer.scala

```diff
@@ -311,6 +308,11 @@ object ActorMaterializerSettings {
 /**
  * This class describes the configurable properties of the [[ActorMaterializer]].
  * Please refer to the `withX` methods for descriptions of the individual settings.
+ *
+ * Important note: `initialInputBufferSize`, `maxInputBufferSize`, `dispatcher` and
+ * `supervisionDecider` must not be used as values in the materializer, or anything the materializer phases use
+ * since these settings allow for overriding using [[Attributes]]. They must always be gotten from the effective
+ * attributes.
```

Comment:
Yeah, I put it in the scaladoc because I thought it would be unnoticeable in a comment, but maybe that isn't strong enough a motivation to include it here.